File: 2021/11/11/OddCells.java
/**
 * https://leetcode.com/problems/cells-with-odd-values-in-a-matrix/
 * @author Andrew Jarombek
 * @since 11/11/2021
 */

public class OddCells {
    public int oddCells(int m, int n, int[][] indices) {
        int[][] matrix = new int[m][n];

        for (int[] location: indices) {
            for (int i = 0; i < matrix[0].length; i++) {
                matrix[location[0]][i] += 1;
            }

            for (int i = 0; i < matrix.length; i++) {
                matrix[i][location[1]] += 1;
            }
        }

        int count = 0;

        for (int[] row : matrix) {
            for (int cell : row) {
                if (cell % 2 == 1) {
                    count++;
                }
            }
        }

        return count;
    }
}

File: 2022/07/31/odd_cells.py
"""
https://leetcode.com/problems/cells-with-odd-values-in-a-matrix/
Author: Andrew Jarombek
Date: 7/31/2022
"""

from typing import List


def odd_cells(m: int, n: int, indices: List[List[int]]) -> int:
    rows = [0] * m
    cols = [0] * n

    for index in indices:
        rows[index[0]] += 1
        cols[index[1]] += 1

    result = 0

    for i in range(m):
        for j in range(n):
            if (rows[i] + cols[j]) % 2 == 1:
                result += 1

    return result


if __name__ == '__main__':
    assert odd_cells(2, 3, [[0, 1], [1, 1]]) == 6
